cmake/codegen.cmake

```diff
@@ -55,10 +58,12 @@ macro(sc_codegen_ex Target SrcPath OutputPath)
             DEPENDS ${HEADER_FILES}
         )
 
-        set (SUB_TARGET "${Target}-GenerateCode")
+        set(SUB_TARGET "${Target}-GenerateCode")
         add_custom_target(${SUB_TARGET} DEPENDS ${CACHE_FILE} SOURCES ${HEADER_FILES})
 
-        add_dependencies(${SUB_TARGET} sc-code-generator)
+        if(TARGET sc-code-generator)
+            add_dependencies(${SUB_TARGET} sc-code-generator)
+        endif()
```

Comment:
If possible, I'd like to implement codegen and other similar targets using [ExternalProject_Add](https://cmake.org/cmake/help/latest/module/ExternalProject.html), but that's a subject of discussions and should be implemented in a separate PR anyway